### DISCUSSION TITLE
Fileupload component maxsize calculation

### DISCUSSION THREAD

**Discussion by RamaKrishna-CEVA** [primefaces/community#544] [PrimeReact] [answered]
Hi Team,

We have updated to primereact version 10.0.9, after updating we have an issue with the Fileupload component max size. we are passing **30000** it should be 30 KB, it was working fine before the update, but after an update, it is showing 29.297 KB, can you please check and resolve this?
please check the attached screenshot for your reference.

reference URL: https://stackblitz.com/edit/react-gwrmvd?file=src%2FApp.js
<img width="684" alt="fileupload" src="https://github.com/primefaces/community/assets/149481901/47209504-4431-45e4-917c-fb047c4659d9">

**Comment by melloware** [comment, marked as answer]
https://en.wikipedia.org/wiki/Kilobyte

>The term 'kilobyte' has traditionally been used to refer to 1024 bytes (210 B).[[5]](https://en.wikipedia.org/wiki/Kilobyte#cite_note-5)[[6]](https://en.wikipedia.org/wiki/Kilobyte#cite_note-6)[[7]](https://en.wikipedia.org/wiki/Kilobyte#cite_note-7) The usage of the metric prefix kilo for binary multiples arose as a convenience, because 1024 is approximately 1000.[[8]](https://en.wikipedia.org/wiki/Kilobyte#cite_note-8)

> The binary interpretation of metric prefixes is still prominently used by the [Microsoft Windows](https://en.wikipedia.org/wiki/Microsoft_Windows) operating system.

and the PrimeReact code is using 1024.

```js
const formatSize = (bytes) => {
            const k = 1024;
            const dm = 3;
            const sizes = localeOption('fileSizeTypes');

            if (bytes === 0) {
                return `0 ${sizes[0]}`;
            }

            const i = Math.floor(Math.log(bytes) / Math.log(k));
            const formattedSize = parseFloat((bytes / Math.pow(k, i)).toFixed(dm));

            return `${formattedSize} ${sizes[i]}`;
        };
```

So basically 29.297 is over 30KB
